tests/stages/arxiv/test_arxiv_source.py

```diff
@@ -0,0 +1,220 @@
+    documents = stage._process_pages(pdf_file)
+
+    assert stage._total_pages == 1
+
+    assert len(documents) == 1
+    document = documents[0]
+    assert document.page_content == 'Morpheus\nunittest'
+    assert document.metadata == {'source': pdf_file, 'page': 0}
+
+
+def test_process_pages_error(config: Config, tmp_path: str):
+    bad_pdf_filename = os.path.join(tmp_path, "bad.pdf")
+    with open(bad_pdf_filename, 'w', encoding='utf-8') as fh:
+        fh.write("Not a PDF")
+
+    stage = ArxivSource(config, query="unittest", cache_dir=tmp_path)
+
+    with pytest.raises(RuntimeError):
+        stage._process_pages(bad_pdf_filename)
+
+
+@mock.patch("langchain.document_loaders.PyPDFLoader")
+def test_process_pages_retry(mock_pdf_loader: mock.MagicMock, config: Config, pypdf: types.ModuleType):
+    call_count = 0
+
+    def mock_load():
+        nonlocal call_count
+        call_count += 1
+        if call_count < 5:
+            raise pypdf.errors.PdfStreamError()
+
+        if call_count == 5:
+            return ["unittest"]
+
+        assert False, "Should not be called more than 5 times"
+
+    mock_pdf_loader.return_value = mock_pdf_loader
+    mock_pdf_loader.load.side_effect = mock_load
+    stage = ArxivSource(config, query="unittest", cache_dir="/does/not/exist")
+    documents = stage._process_pages('/does/not/exist/fake.pdf')
+
+    assert stage._total_pages == 1
+    assert call_count == 5
+
+    assert documents == ["unittest"]
+
+
+@pytest.mark.parametrize("chunk_size", [200, 1000])
+def test_splitting_pages(config: Config,
+                         pdf_file: str,
+                         langchain: types.ModuleType,
+                         chunk_size: int,
+                         dataset_cudf: DatasetManager):
+    chunk_overlap = 100
+    content = "Morpheus\nunittest"
+    while len(content) < chunk_size:
+        content += f'\n{string.ascii_lowercase}'
+
+    # The test PDF is quite small, so we need to pad it out to get multiple pages
+    # I checked this won't evently divide into either of the two values for chunk_size
+    if len(content) != chunk_size:
+        split_point = content.rfind('\n')
+        first_row = content[0:split_point]
+        second_row = content[split_point + 1:]
+        while len(second_row) < chunk_overlap:
+            split_point = content[0:split_point].rfind('\n')
+            second_row = content[split_point + 1:]
+
+        page_content_col = [first_row, second_row]
+    else:
+        page_content_col = [content]
+
+    num_expected_chunks = len(page_content_col)
+    source_col = []
+    page_col = []
+    for _ in range(num_expected_chunks):
+        source_col.append(pdf_file)
+        page_col.append(0)
+
+    expected_df = cudf.DataFrame({"page_content": page_content_col, "source": source_col, "page": page_col})
+
+    loader = langchain.document_loaders.PyPDFLoader(pdf_file)
+    documents = loader.load()
```

Comment:
This looks more like we're testing the behavior of an external library vs the ArxivSource functionality.

Reply:
In this case I needed a list of `langchain.schema.Document` instances to test the functionality of the `_splitting_pages` method.